### PR TITLE
Levels of Detail

### PR DESCRIPTION
**Pull request by erlewa**
Updated load.rs:
- Meshing algorithm now returns a Vec<BufferStorage> with each entry in the vec being a different level of detail
- Flowed this change down into other files and structs

Updated geometry.rs:
- generate_buffers now generates 3 lod_layers instead of a blend_layer
- Created a BufferLayer struct to contain the vertex and index buffer and num_indices for each group of buffers
- Geometry now contains BufferLayers instead of straight buffers

Updated state.rs:
- Added logic to check the current camera zoom and select the LOD to render based on this value

### CODE REVIEW

File: heatmap-client/src/canvas/geometry.rs [hunk 1]
```diff
@@ -25,45 +25,96 @@ const COLOR_VERTICES: &[Vertex] = &[
...
+        render_context: &RenderContext,
+        buffer_data: Vec<BufferStorage>,
+    ) -> Self {
         //////////////////////////////
         // Set up buffers to render //
         //////////////////////////////
 
-        let blend_vertex_buffer =
+        // LOD 0
+        let lod_0_vertex_buffer =
```
Comment (Id405) [inline heatmap-client/src/canvas/geometry.rs]: It might make sense to store these all in a vec, allowing for a variable amount of LOD levels

Comment (Id405) [reply heatmap-client/src/canvas/geometry.rs]: and it would reduce code duplication

**Review by Id405 — DISMISSED** [review]

File: heatmap-client/src/canvas/geometry.rs (hunk 1 above)
Comment (erlewa) [reply heatmap-client/src/canvas/geometry.rs]: This is done!

Comment (Id405) [reply heatmap-client/src/canvas/geometry.rs]: you can generate the lod vertex buffers themselves using a map statement to further reduce code duplication if you want

**Review by Id405 — DISMISSED** [review]

File: heatmap-client/src/canvas/geometry.rs [hunk 2]
```diff
@@ -25,45 +24,62 @@ const COLOR_VERTICES: &[Vertex] = &[
...
-                    label: Some("Blending Vertex Buffer"),
-                    contents: bytemuck::cast_slice(buffer_data.vertices.as_slice()),
-                    usage: wgpu::BufferUsages::VERTEX,
-                });
+        for i in 0..buffer_data.len() {
+            let lod_vertex_buffer =
+                render_context
+                    .device
+                    .create_buffer_init(&wgpu::util::BufferInitDescriptor {
+                        label: Some(&("LOD ".to_string() + i.to_string().as_str() + " Vertex Buffer")),
```
Comment (Id405) [inline heatmap-client/src/canvas/geometry.rs]: use the format!() macro

**Review by Id405 — DISMISSED** [review]

File: heatmap-client/src/canvas/geometry.rs (hunk 2 above)
Comment (erlewa) [reply heatmap-client/src/canvas/geometry.rs]: Real, one sec!

**Review by Id405 — APPROVED** [review]

**Pull request merged**